### PR TITLE
services/jsonrpc: do not throw away engine error details

### PR DESCRIPTION
**Pull request by jchappelow**
Resolves https://github.com/kwilteam/kwil-db/issues/725

The following rpc service methods calling into engine are now returning the full error when it is unidentified:
- `Query` (this is the most sensible place to return the full engine error)
- `ListDatabases` (but this error is more likely to be genuinely internal, since an empty slice is the result if none found rather than an error)
- `Call` (like `Query` it is likely that the "internal" error will be important to help the user resolve the issue with their call or the dataset definition itself).

Note that `Schema` also uses the `engineError` helper, but `engine.GetSchema` only returns `ErrDatasetNotFound` or `nil`, and there is no internal error that would have been discarded.

### CODE REVIEW

**Comment by jchappelow** [comment]
**Reference** (no error):

```
$  ./kwil-cli database query -i x6f89a49f1bd5fd49b7f4e13fa9cdb76eb05cae29aa776cd88a9da0cf "SELECT * FROM users WHERE age > 25"
| age |         id          | username |                   wallet                   |
+-----+---------------------+----------+--------------------------------------------+
|  42 | 1061840862069797577 | me       | 0x5a9fd9d33B310eA8Fb192818C1016A353e734C41 |
```

Bad query ("agex" instead of "age"):

**Before** 

```
$  ./kwil-cli database query -i x6f89a49f1bd5fd49b7f4e13fa9cdb76eb05cae29aa776cd88a9da0cf "SELECT * FROM users WHERE agex > 25"
error querying database: jsonrpc.Error: code = -300, message = internal engine error, data = null
err code = -300, msg = internal engine error
```

**After**

```
 $  ./kwil-cli database query -i x6f89a49f1bd5fd49b7f4e13fa9cdb76eb05cae29aa776cd88a9da0cf "SELECT * FROM users WHERE agex > 25"
error querying database: jsonrpc.Error: code = -300, message = ERROR: column "agex" does not exist (SQLSTATE 42703), data = null
err code = -300, msg = ERROR: column "agex" does not exist (SQLSTATE 42703)
```

tldr: `msg = internal engine error` => `ERROR: column "agex" does not exist (SQLSTATE 42703)`

Which for an ad hoc query is exactly what I'd want in return.

**Review by Yaiba — APPROVED** [review]

**Review by brennanjl — APPROVED** [review]
